crates/rust-analyzer/src/flycheck.rs

```diff
@@ -183,7 +183,7 @@ pub(crate) enum Progress {
 }
 
 enum StateChange {
-    Restart { package: Option<String>, saved_file: Option<AbsPathBuf> },
+    Restart { package: Option<String>, saved_file: Option<AbsPathBuf>, target: Option<String> },
```

Comment:
```suggestion
    Restart { package: Option<String>, saved_file: Option<AbsPathBuf>, target: Option<Target>},
```
```rs
enum Target {
    Bin(String)
}
```
would be better to change that to an enum up front so we can support other targets more easily (mainly examples come to mind), as those require different flags than bin

Reply:
Bin, example, test and bench